test/primus.test.js

```diff
@@ -104,6 +104,22 @@ describe('Primus', function () {
 
     throw new Error('I should have throwed above');
   });
+  
+  it('accepts a third-party spark id generator', function () {
+    function testIdGenerator() {
+      return 'test_'+(Math.random()+1).toString(36).substring(7);
```

Comment:
Yes, that's probably sufficient. I'll push a new commit momentarily with your recommended changes.